modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/ExtensionDefinitionParser.java

```diff
@@ -743,71 +743,90 @@ private boolean isExpressionFunction(MetadataType metadataType)
+            {
+                constructedValue = LocalDateTime.of(dateTime.getYear(),
+                                                    dateTime.getMonthOfYear(),
+                                                    dateTime.getDayOfMonth(),
+                                                    dateTime.getHourOfDay(),
+                                                    dateTime.getMinuteOfHour(),
+                                                    dateTime.getSecondOfMinute());
+
+            }
+            else if (type.equals(Calendar.class))
+            {
+                Calendar calendar = Calendar.getInstance();
+                calendar.setTime(dateTime.toDate());
+                constructedValue = calendar;
+            }
+
+            if (constructedValue == null)
+            {
+                throw new IllegalArgumentException(format("Could not construct value of type '%s' from String '%s'", type.getName(), value));
+            }
+            else
+            {
+                value = constructedValue;
+            }
         }
-        else
+
+        if (value instanceof Date || value instanceof LocalDate || value instanceof LocalDateTime || value instanceof Calendar)
         {
-            Date date = doParseDate(value, CALENDAR_FORMAT, defaultValue);
-            Calendar calendar = Calendar.getInstance();
-            calendar.setTime(date);
-            return new StaticValueResolver(calendar);
+            return new StaticValueResolver(value);
         }
+
+        throw new IllegalArgumentException(format("Could not transform value of type '%s' to a valid date type", value != null ? value.getClass().getName() : "null"));
     }
 
-    private ValueResolver parseDate(Object value, DateType dateType, Object defaultValue)
+    private static DateTime getParsedDateTime(String value)
     {
-        if (isExpression(value, parser))
+        try
         {
-            return new TypeSafeExpressionValueResolver((String) value, getType(dateType));
+            return new DateTime(value);
         }
-        else
+        catch (DateTimeParseException e)
         {
-            return new StaticValueResolver(doParseDate(value, DATE_FORMAT, defaultValue));
+            throw new IllegalArgumentException(format("Could not parse value '%s' according to ISO 8601", value));
         }
     }
 
-    private Date doParseDate(Object value, String parseFormat, Object defaultValue)
+    private ValueResolver parseDate(Object value, MetadataType dateType, Object defaultValue)
     {
-        if (value == null)
-        {
-            if (defaultValue == null)
-            {
-                return null;
-            }
 
-            value = defaultValue;
+        Class<?> type = getType(dateType);
+        if (isExpression(value, parser))
+        {
+            return new TypeSafeExpressionValueResolver((String) value, type);
         }
 
-        if (value instanceof String)
+        if (value == null)
```

Comment:
this is wrong. Nothing of this makes sense if the value is not an String. You should keep the original logic with the only exception of the SimpleDateFormat thing